giraph-core/src/main/java/org/apache/giraph/writable/kryo/KryoWritableWrapper.java

```diff
@@ -120,4 +126,50 @@ public static Writable wrapIfNeeded(Object object) {
   public static <T> T wrapAndCopy(T object) {
     return WritableUtils.createCopy(new KryoWritableWrapper<>(object)).get();
   }
+
+  /**
+   * Try converting the object to byte array.
+   * @param object Object
+   * @param <T> Type
+   * @return byte array
+   */
+  public static <T> byte [] tryConvertToByteArray(T object) {
+    byte [] arr = null;
+    try {
+      KryoWritableWrapper<T> wrapper =
+              new KryoWritableWrapper<>(object);
+      arr = WritableUtils.toByteArrayUnsafe(wrapper);
+      // Checkstyle exception due to unsafe conversion
+      // CHECKSTYLE: stop IllegalCatch
+    } catch (Exception e) {
+      // CHECKSTYLE: resume IllegalCatch
+      LOG.error("Failed to convert to byte array: " +
+              ExceptionUtils.getStackTrace(e));
+    }
+    return arr;
+  }
+
+  /**
+   * Try converting from byte array
+   * @param arr byte array
+   * @param <T> type
+   * @return original object
+   */
+  public static <T> T tryConvertFromByteArray(byte [] arr) {
+    T result = null;
+    try {
+      KryoWritableWrapper<T> wrapper =
+              new KryoWritableWrapper<>();
+      WritableUtils.fromByteArrayUnsafe(
```

Comment:
Any reason you chose to use unsafe instead of simple writeToByteArray/readFromByteArray?

Reply:
Agreed.